### PR TITLE
remove mp_read_option_raw

### PR DESCRIPTION
**Pull request by Dudemanguy**
Second go at this. This time it gets rid of everything. Bunch of commits so it's more readable (hopefully).

### CODE REVIEW

File: DOCS/man/options.rst [hunk 1]
```diff
@@ -3481,7 +3481,7 @@ Window
 Disc Devices
 ------------
 
-``--cdrom-device=<path>``
+``--cdda-device=<path>``
     Specify the CD-ROM device (default: ``/dev/cdrom``).
```
Comment (sfan5) [inline DOCS/man/options.rst:3485]: not sure what the most accurate terminology is here
```suggestion
    Specify the CD device for CDDA playback (default: ``/dev/cdrom``).
```

File: options/options.c [hunk 2]
```diff
@@ -843,13 +856,11 @@ static const m_option_t mp_opts[] = {
...
 #if HAVE_GL_WIN32
-    {"opengl-dwmflush", OPT_CHOICE(wingl_dwm_flush,
-        {"no", -1}, {"auto", 0}, {"windowed", 1}, {"yes", 2})},
+    {"", OPT_SUBSTRUCT(wingl_opts, wingl_conf)},
 #endif
 
 #if HAVE_CUDA_HWACCEL
-    {"cuda-decode-device", OPT_CHOICE(cuda_device, {"auto", -1}),
-        M_RANGE(0, INT_MAX)},
+    {"", OPT_SUBSTRUCT(cuda_opts, cuda_conf)},
```
Comment (sfan5) [inline options/options.c:863]: could use `"cuda"` as prefix here

File: DOCS/man/options.rst (hunk 1 above)
Comment (Dudemanguy) [reply DOCS/man/options.rst:3485]: Sounds good to me.

**Comment by Dudemanguy** [comment]
I was able to drop the user_filters commit thanks to #12436.

**Review by sfan5 — APPROVED** [review]
not tested or anything

**Pull request merged**